packages/compass-indexes/src/modules/indexes.ts

```diff
@@ -225,3 +227,79 @@ export const dropFailedIndex = (
     void dispatch(fetchIndexes());
   };
 };
+
+export const hideIndex = (
+  indexName: string
+): ThunkAction<Promise<void>, RootState, void, AnyAction> => {
+  return async (dispatch, getState) => {
+    const { dataService, namespace } = getState();
+    const confirmed = await showConfirmation({
+      dataTestId: 'hide-index-confirmation-modal',
+      title: `Hiding \`${indexName}\``,
+      description: createElement(Fragment, {
+        children: [
+          'The index `',
+          createElement('b', {
+            children: indexName,
+          }),
+          '` will no longer be visible to the query planner and cannot be used to support a query. If the impact is negative, you can unhide this index.',
+        ],
+      }),
+    });
```

Comment:
You can just rename this file to tsx if you need to use jsx here

Reply:
Yea I was actually thinking on this a lot but then recalled the horrible diffs (file-deleted and a new-file added). Is that still a thing? I didn't try it since last time I saw that.

Reply:
I think the initial inconvenience of this is surpassed by reading this code some time later 😄 If you really don't want to rename the file, I saw Rhys moving all jsx including code to a separate tsx file and importing from there, maybe a good enough compromise between two approaches?

Reply:
That sounds really good to me. I will do that right away. 😄